lib/widgets/message_list.dart

```diff
@@ -849,61 +852,75 @@ class MessageWithPossibleSender extends StatelessWidget {
                       userId: message.senderId)),
-                  child: Avatar(size: 35, borderRadius: 4,
-                    userId: message.senderId)))
-            : const SizedBox(width: 3 + 35 + 11),
-          Expanded(
-            child: Column(
-              crossAxisAlignment: CrossAxisAlignment.stretch,
-              children: [
-                if (item.showSender) ...[
-                  const SizedBox(height: 3),
-                  GestureDetector(
-                    onTap: () => Navigator.push(context,
-                      ProfilePage.buildRoute(context: context,
-                        userId: message.senderId)),
-                    child: Text(message.senderFullName, // TODO get from user data
-                      style: const TextStyle(fontWeight: FontWeight.bold))),
-                  const SizedBox(height: 4),
-                ],
-                MessageContent(message: message, content: item.content),
-                if ((message.reactions?.total ?? 0) > 0)
-                  ReactionChipsList(messageId: message.id, reactions: message.reactions!)
-              ])),
-          Container(
-            width: 80,
-            padding: const EdgeInsets.only(top: 4, right: 16 - 8),
-            alignment: Alignment.topRight,
-            child: Text(time, style: _kMessageTimestampStyle)),
+                  child: Row(
+                    children: [
+                      Avatar(size: 32, borderRadius: 3,
+                        userId: message.senderId),
+                      const SizedBox(width: 8),
+                      Flexible(
+                        child: Text(message.senderFullName, // TODO get from user data
+                          style: const TextStyle(
+                            fontFamily: 'Source Sans 3',
+                            fontSize: 18,
+                            height: (22 / 18),
+                          ).merge(weightVariableTextStyle(context, wght: 600,
+                                    wghtIfPlatformRequestsBold: 900)),
+                          overflow: TextOverflow.ellipsis)),
+                    ]))),
+              Text(
+                _kMessageTimestampFormat.format(
+                  DateTime.fromMillisecondsSinceEpoch(1000 * message.timestamp)),
+                style: TextStyle(
+                  color: _kMessageTimestampColor,
+                  fontFamily: 'Source Sans 3',
+                  fontSize: 16,
+                  height: (18 / 16),
+                  fontFeatures: const [FontFeature.enable('c2sc'), FontFeature.enable('smcp')],
+                ).merge(weightVariableTextStyle(context))),
+            ]))
+      : null;
+
+    return GestureDetector(
+      behavior: HitTestBehavior.translucent,
+      onLongPress: () => showMessageActionSheet(context: context, message: message),
+      child: Padding(
+        padding: const EdgeInsets.symmetric(vertical: 4),
+        child: Column(children: [
+          if (senderRow != null) senderRow,
```

Comment:
I think the layout here is somewhat easier to read and understand if the padding around senderRow stays here:
```suggestion
          if (senderRow != null)
            Padding(padding: const EdgeInsets.fromLTRB(16, 2, 16, 4),
              child: senderRow),
```

That way the "16" here is just a couple of lines away from the "16" that gives the widths of the gutters around the message body, below. And similarly the vertical 2px and 4px are close by the 4px vertical padding around the whole thing, which they interact with closely.

(The version I've given also contains a smaller change: the use of `EdgeInsets.fromLTRB` is helpful for keeping things compact.)